### PR TITLE
Upgrade findbugs to spotbugs

### PR DESCRIPTION
**Pull request by johnlcox**
What was changed? Why is this necessary?
----------------------------------------

Upgrades from findbugs to spotbugs. Spotbugs is the successor to findbugs.

How was it tested?
----

Ran `mvn clean install`

How to test
----

*This is bare minimum acceptable testing*

- [ ] `mvn clean install -U`

Reviewers
----

- [ ] [John Leacox](https://github.com/johnlcox)
- [ ] [Sundeep Paruvu](https://github.com/sparuvu)
- [ ] [Nimesh Subramanian](https://github.com/nimeshsubramanian)
- [ ] [Supriya Lal](https://github.com/lal-s)
- [ ] [Brian van de Boogaard](https://github.com/b-boogaard)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -743,14 +726,14 @@
...
         </plugin>
         <plugin>
-          <groupId>org.codehaus.mojo</groupId>
-          <artifactId>findbugs-maven-plugin</artifactId>
+          <groupId>com.github.spotbugs</groupId>
+          <artifactId>spotbugs-maven-plugin</artifactId>
           <version>3.0.5</version>
           <configuration>
             <effort>Max</effort>
             <threshold>Default</threshold>
```
Comment (shiftyeyes) [inline pom.xml]: Default is the default, this can be removed.

File: pom.xml [hunk 2]
```diff
@@ -91,17 +91,18 @@
     <governator.version>1.17.4</governator.version>
     <jackson.version>2.9.2</jackson.version>
```
Comment (shiftyeyes) [inline pom.xml]: /docs/pom.xml, Can you update sphinx-maven-plugin to 1.7.0. 1.6.1.Final doesn't exist at https://repo.maven.apache.org/maven2/kr/motd/maven/sphinx-maven-plugin/

File: pom.xml [hunk 3]
```diff
@@ -743,14 +726,14 @@
           </configuration>
         </plugin>
         <plugin>
-          <groupId>org.codehaus.mojo</groupId>
-          <artifactId>findbugs-maven-plugin</artifactId>
+          <groupId>com.github.spotbugs</groupId>
+          <artifactId>spotbugs-maven-plugin</artifactId>
           <version>3.0.5</version>
```
Comment (johnlcox) [inline pom.xml]: Should use the `${spotbugs.plugin.version}` here as well.

File: pom.xml (hunk 2 above)
Comment (johnlcox) [reply pom.xml]: https://github.com/cerner/beadledom/commit/66d28e32652a3d9b720293986ee67e0fead66a99?diff=unified#diff-48fabae9c2e321b2ab4d99057d5dc32eL46

File: pom.xml (hunk 3 above)
Comment (johnlcox) [reply pom.xml]: https://github.com/cerner/beadledom/commit/66d28e32652a3d9b720293986ee67e0fead66a99?diff=unified#diff-600376dffeb79835ede4a0b285078036L731

File: pom.xml (hunk 1 above)
Comment (johnlcox) [reply pom.xml]: https://github.com/cerner/beadledom/commit/66d28e32652a3d9b720293986ee67e0fead66a99?diff=unified#diff-600376dffeb79835ede4a0b285078036L734

File: spotbugs-exclude.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+<FindBugsFilter>
```
Comment (sparuvu) [inline spotbugs-exclude.xml]: nit: should we add the DTD ? to make it a valid xml file.

```xml 
<?xml version="1.0" encoding="UTF-8"?>
 <FindBugsFilter
              xmlns="https://github.com/spotbugs/filter/3.0.0"
              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
              xsi:schemaLocation="https://github.com/spotbugs/filter/3.0.0 https://raw.githubusercontent.com/spotbugs/spotbugs/3.1.0/spotbugs/etc/findbugsfilter.xsd">
```

Comment (sparuvu) [reply spotbugs-exclude.xml]: sadly they haven't changed the root of the xml to spotbugsfilter :(

Comment (johnlcox) [reply spotbugs-exclude.xml]: https://github.com/cerner/beadledom/commit/3dcd2eabc4d96ed1c9547a1bffc19da4ecf92574

**Pull request merged**